docs/docs/concepts/advanced/private_execution_environment.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Private Execution Environment (PXE)
+---
+
+The Private Execution Environment (or PXE, pronounced 'pixie') is a client-side library for the execution of private operations. It is a TypeScript library and can be run within Node, such as when you run the sandbox, within the browser, or any other environment in which TypeScript can run. For example, in future it could be run inside wallet software.
+
+The PXE generates proofs of private function execution, and sends these proofs along with public function requests to the sequencer. Private inputs never leave the client-side PXE.
+
+```mermaid
+graph TD;
+
+    subgraph client[Client]
+        subgraph pxe [PXE]
+            acirSim[ACIR Simulator]
+            db[Database]
+            keyStore[KeyStore]
+        end
+    end
+
+    subgraph server[Server]
```

Comment:
Is this called a server? To me this implies it is running off of the users' device, which isn't correct?

Reply:
the PXE service is run server-side, the PXE is run client-side. the service provides the APIs for the PXE. the PXE itself acts as a sort of relay between the PXE service and a dapp. is the graph unclear?

Reply:
In the graph, it isn't clear where the Server is running. It would be helpful to clarify that the PXE server is run by an Aztec dapp and that user's client-side PXE communicates with the PXE server which is run by the application that the user is interacting with.

Reply:
```suggestion
    subgraph server[Application Server]
```